autouri/cli.py

```diff
@@ -0,0 +1,210 @@
+        'cp',
+        help='AutoURI(src).cp(target): Copy source to target. '
+             'target must be a full filename/directory. '
+             'Target directory must have a trailing directory separator '
+             '(e.g. /)',
+        parents=[parent_src, parent_target, parent_cp, parent_all])
+
+    p_read = subparser.add_parser(
+        'read',
+        help='AutoURI(src).read(): Read from source.',
+        parents=[parent_src, parent_all])
+
+    p_write = subparser.add_parser(
+        'write',
+        help='AutoURI(src).write(text): Write text on source.',
+        parents=[parent_src, parent_all])
+    p_write.add_argument('text',
+        help='Text to be written to source file.')
+
+    p_rm = subparser.add_parser(
+        'rm',
+        help='AutoURI(src).rm(): Delete source.',
+        parents=[parent_src, parent_all])
+
+    p_loc = subparser.add_parser(
+        'loc',
+        help='AutoURI(src).localize_on(target): Localize source on target directory '
+             'Target directory must end with directory separator',
+        parents=[parent_src, parent_target, parent_cp, parent_all])
+    p_loc.add_argument('--recursive', action='store_true',
+        help='Recursively localize source into target directory.')
+
+    p_presign = subparser.add_parser(
+        'presign',
+        help='AutoURI(src).get_presigned_url(). For cloud-based URIs only.',
+        parents=[parent_src, parent_all])
+    p_presign.add_argument('--gcp-private-key-file',
+        help='GCP private key file (JSON format) to presign gs:// URIs.')
+    p_presign.add_argument('--duration', type=int,
+        help='Duration of presigned URL in seconds.')
+
+    if len(sys.argv[1:]) == 0:
+        parser.print_help()
+        parser.exit()
+
+    args = parser.parse_args()
+    if args.version:
+        print(version)
+        parser.exit()
+
+    if args.debug:
+        log_level = 'DEBUG'
+    else:
+        log_level = 'INFO'
+    logging.basicConfig(
+        level=log_level,
+        format='%(asctime)s|%(name)s|%(levelname)s| %(message)s')
+    # suppress filelock logging
+    logging.getLogger('filelock').setLevel('CRITICAL')
+
+    return args
+
+
+def get_local_path_if_valid(s):
+    abspath = os.path.abspath(os.path.expanduser(s))
+    dirname = os.path.dirname(abspath)
+    if os.path.isdir(abspath):
+        return abspath + os.sep
+    elif os.path.exists(abspath):
+        return abspath
+    elif os.path.exists(dirname):
+        return abspath
+    return s
+
+
+def main():
+    args = parse_args()
+
+    src = get_local_path_if_valid(args.src)
+
+    if args.action in ('cp', 'loc'):
+        if args.use_gsutil_for_s3:
+            GCSURI.init_gcsuri(use_gsutil_for_s3=True)
+        target = get_local_path_if_valid(args.target)
+
+    if args.action == 'metadata':
+        m = AutoURI(src).get_metadata()
+        print(m)
+
+    elif args.action == 'cp':
+        u_src = AutoURI(src)
+        _, flag = u_src.cp(
```

Comment:
I think I suggested this earlier, that in perfect world `cp` would return some sort of `CopyResult` object that would hide the return codes. Not a required change.

Reply:
Sure, made a ticket PIP-1039 for it. Will be working on that later.